lexer_test.go

```
@@ -0,0 +1,119 @@
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			tc.lexer.Init(tc.input)
+			opts := []cmp.Option{
+				cmp.AllowUnexported(SimpleToken{}, EagerLexer{}),
+				cmpopts.IgnoreFields(EagerLexer{}, "lexFunc"),
+			}
+			if diff := cmp.Diff(tc.want, tc.lexer, opts...); diff != "" {
+				t.Fatalf(diff)
```

golangci-lint (1.62.2)

printf: non-constant format string in call to (*testing.common).Fatalf

(govet)